sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -214,11 +215,13 @@ case class Invoke(
   override def eval(input: InternalRow): Any =
     throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
 
+  private lazy val encodedFunctionName = TermName(functionName).encodedName.toString
```

Comment:
does `StaticInvoke` have some issue?

Reply:
Maybe, although I didn't have concrete case causing the issue for now.

Reply:
Since we use `Invoke` to access field(s) in object, this can be an issue. I didn't found `StaticInvoke` used similarly. So it should be fine.